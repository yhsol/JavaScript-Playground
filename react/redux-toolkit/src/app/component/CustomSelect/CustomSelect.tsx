import React from "react";

interface CustomSelectProps {
  onChange: (value: string) => void;
  options: string[];
  value: string;
  label: string;
  idPrefix: string;
}

const CustomSelect: React.FC<CustomSelectProps> = ({
  options,
  value,
  onChange,
  label,
  idPrefix,
}) => {
  const selectId = `${idPrefix}-select`;
  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>): void =>
    onChange(event?.target.value);
  return (
    <div>
      <label htmlFor={selectId}>{label}</label>
      <select
        id={selectId}
        value={value}
        className="form-control"
        onChange={handleSelect}
      >
        {options.map((option) => (
          <option key={option}>{option}</option>
        ))}
      </select>
    </div>
  );
};

export default CustomSelect;
