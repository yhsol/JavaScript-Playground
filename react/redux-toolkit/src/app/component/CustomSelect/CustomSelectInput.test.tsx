import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import CustomSelect from "./CustomSelect";

const changeHandler = jest.fn();

const options = ["option 1", "option 2"];

test("renders the text input", () => {
  const { rerender, asFragment } = render(
    <CustomSelect
      onChange={changeHandler}
      value={"option 2"}
      options={options}
      label={"My Label"}
      idPrefix={"myTest"}
    />
  );
  expect(asFragment()).toMatchInlineSnapshot(`
    <DocumentFragment>
      <div>
        <label
          for="myTest-select"
        >
          My Label
        </label>
        <select
          class="form-control"
          id="myTest-select"
        >
          <option>
            option 1
          </option>
          <option>
            option 2
          </option>
        </select>
      </div>
    </DocumentFragment>
  `);
  const textInput = screen.getByLabelText("My Label");
  expect(textInput).toHaveValue("option 2");
  fireEvent.change(textInput, { target: { value: "option 1" } });
  expect(changeHandler).toHaveBeenCalledWith("option 1");
  rerender(
    <CustomSelect
      onChange={changeHandler}
      options={options}
      value={"option 1"}
      label={"My Label"}
      idPrefix={"myTest"}
    />
  );
  expect(textInput).toHaveValue("option 1");
});
