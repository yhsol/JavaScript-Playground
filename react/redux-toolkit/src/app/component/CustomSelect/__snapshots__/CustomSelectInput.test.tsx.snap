// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the text input: 
      <DocumentFragment>
          <div>
              <label for="myTest-select">My Label</label>
              <select class="form-control" id="myTest-select">
                  <option>
                      option 1
                  </option>
                  <option>
                      option 2
                  </option>
              </select>
          </div>
      </DocumentFragment>
   1`] = `
<DocumentFragment>
  <div>
    <label
      for="myTest-select"
    >
      My Label
    </label>
    <select
      class="form-control"
      id="myTest-select"
    >
      <option>
        option 1
      </option>
      <option>
        option 2
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`renders the text input: 
    <DocumentFragment>
      <div>
        <label
          for="myTest-select"
        >
          My Label
        </label>
        <select
          class="form-control"
          id="myTest-select"
        >
          <option>
            option 1
          </option>
          <option>
            option 2
          </option>
        </select>
      </div>
    </DocumentFragment>
   1`] = `
<DocumentFragment>
  <div>
    <label
      for="myTest-select"
    >
      My Label
    </label>
    <select
      class="form-control"
      id="myTest-select"
    >
      <option>
        option 1
      </option>
      <option>
        option 2
      </option>
    </select>
  </div>
</DocumentFragment>
`;
